src/models/common_decoder.h

```diff
@@ -530,13 +530,67 @@ class CommonDecoder : public AbstractDecoder {
         TimeLine t("Decoder.forward");
         TimeLine t1("Decoder.embedding");
 
+        // Prepare input
+        int totInputSeqLen = 0;
+        std::vector<int> allInputIds;
+        for (auto seq : seqs) {
+            // TODO: maybe not correct
+            totInputSeqLen += seq->getInputSeqLen();
```

Comment:
Assume input is prompt+past_generated_tokens.
getInputSeqLen() gets the size of prompt, getLatestToken() return the past_generated_tokens.

Reply:
We can check the step by getStep(), if step ==0,  Len = getInputSeqLen(), input token is getInputTokens. else Len is 1 and input is the latest element of getLatestToken().

Reply:
And now we assume all sequences are all prompts(step==0) or all decodes(step>0) like vllm < v0.4. After v0.4, vllm supports mixed calculations between prompts and decodes since April 11...

Reply:
> We can check the step by getStep(), if step ==0, Len = getInputSeqLen(), input token is getInputTokens. else Len is 1 and input is the latest element of getLatestToken().

For verification, the length is not 1.

Reply:
> > We can check the step by getStep(), if step ==0, Len = getInputSeqLen(), input token is getInputTokens. else Len is 1 and input is the latest element of getLatestToken().
> 
> For verification, the length is not 1.

We need to add a new func member in the seq class.